test/networkd-test.py

```diff
@@ -594,6 +594,162 @@ def test_dropin(self):
         self.assertIn('nameserver 192.168.42.1\n', contents)
         self.assertIn('nameserver 127.0.0.1\n', contents)
 
+
+class UnmanagedClientTest(ClientTestBase, unittest.TestCase):
```

Comment:
Please don't subclass `ClientTestBase` for adding these tests -- this is meant to test networkd (client side) against different "remote" servers (dnsmasq or networkd in DHCP server mode), and all these `skip("Inherited")` are ugly.

Either add the new tests to the existing `NetworkdClientTest` class (no need to run them for dnsmasq too), or keep it as a new class but directly derive from `unittest.TestCase`. If there is much functionality that you need from `ClientTestBase` it is also fine to only keep the utility functions in there, and move the actual shared tests into an intermediate `CommonTests(ClientTestBase)` class and have `{Dnsmasq,Networkd}ClientTests` inherit from `CommonTests`.

That said, no need to change the tests before agreeing to the actual design of course (that still seems to be disputed).